Makefile

```
@@ -234,11 +242,12 @@ compose:
 
 #<<<
 compose-%:
+	printf "\n\nErrors during compose-$*: \n" >> $(ERRORLOG)
 	echo "    started compose-$*" >> $(LOGFILE)
-	-source lib/$*/$*_preflight.sh
+	-source lib/$*/$*_preflight.sh 1> >(tee -a >(grep -C 2 "[Ee]rror|[Ww]arning" >> $(ERRORLOG)) /dev/null) 2> >(tee -a $(ERRORLOG))
```

Prefer `-i` to including both capitalizations like this, since all-caps error/warning messages are also common. In addition, warnings often have just `WARN` in them.
```suggestion
	-source lib/$*/$*_preflight.sh 1> >(tee -a >(grep -iC 2 "error|warn" >> $(ERRORLOG)) /dev/null) 2> >(tee -a $(ERRORLOG))
```
I'm also unsure about storing only the lines matching error/warn here -- it might be best to store *everything*, and then grep for errors afterwards (during post-build).